tests/bundle/Routing/ExpressionLanguage/ContentTypeHeaderMatcherFactoryTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Tests\Bundle\Rest\Routing\ExpressionLanguage;
+
+use Closure;
+use Ibexa\Bundle\Rest\Routing\ExpressionLanguage\ContentTypeHeaderMatcherFactory;
+use Ibexa\Contracts\Rest\Input\ParsingDispatcher;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\HttpFoundation\Request;
+
+final class ContentTypeHeaderMatcherFactoryTest extends TestCase
+{
+    private readonly ParsingDispatcher&MockObject $parsingDispatcher;
+
+    private readonly Closure $contentTypeHeaderMatcher;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->parsingDispatcher = $this->createMock(ParsingDispatcher::class);
+        $contentTypeHeaderMatcherFactory = new ContentTypeHeaderMatcherFactory($this->parsingDispatcher);
+        $this->contentTypeHeaderMatcher = $contentTypeHeaderMatcherFactory();
+    }
+
+    public function testRequestContentTypeMatchesRoute(): void
+    {
+        $request = new Request();
+        $request->headers->add([
+            'Content-Type' => 'application/vnd.ibexa.api.CopyContentTypeInput+json',
+        ]);
+
+        $closure = $this->contentTypeHeaderMatcher;
+        $this->mockParsingDispatcher('application/vnd.ibexa.api.CopyContentTypeInput');
+
+        self::assertTrue($closure($request, 'application/vnd.ibexa.api.CopyContentTypeInput'));
+    }
+
+    public function testRequestContentTypeDoesNotMatchRoute(): void
+    {
+        $request = new Request();
+        $request->headers->add([
+            'Content-Type' => 'application/vnd.ibexa.api.CreateContentTypeInput+xml',
+        ]);
+
+        $closure = $this->contentTypeHeaderMatcher;
+        $this->mockParsingDispatcher('application/vnd.ibexa.api.CreateContentTypeInput');
+
+        self::assertFalse($closure($request, 'application/vnd.ibexa.api.CopyContentTypeInput'));
+    }
```

Comment:
These tests suggest that they are checking the ability of `ParsingDispatcher` to recognize media types. They do not, since your `$this->parsingDispatcher` is a mock that can have it's matching turn on and off on demand. So you're in practice only testing that your mock is actually reached (`->expects(self::once())`).

Either use a proper ParsingDispatcher instance (since it's an implementation, not an interface - and does not have direct dependencies either), or move these to `ParsingDispatcherTest` and actually test that `ParsingDispatcher::fetchMediaTypeWithoutVersion()` it behaves as you want it to.

Alternatively, add more test cases that work against an actual instance.